cibyl/sources/elasticsearch/api.py

```diff
@@ -355,6 +354,98 @@ def get_deployment(self, **kwargs):
 
         return AttributeDictValue("jobs", attr_type=Job, value=job_objects)
 
+    @speed_index({'base': 3})
+    def get_tests(self, **kwargs):
+        """
+            Get tests for a elasticsearch job.
+
+            :returns: container of jobs with the last completed build
+            (if any) and the tests
+            :rtype: :class:`AttributeDictValue`
+        """
+        jobs_found = self.get_jobs(**kwargs)
+        query_body = {
+            "query": {
+                "bool": {
+                    "must": [
+                        {},
+                        {
+                            "exists": {
+                                "field": "test_status"
+                            }
+                        },
+                        {
+                            "exists": {
+                                "field": "test_name"
+                            }
+                        }
+                    ]
+                }
+            },
+            "sort": [{"timestamp.keyword": {"order": "desc"}}]
+        }
+
+        results = []
+        hits = []
+        for job in jobs_found:
+            query_body['query']['bool']['must'][0] = {
+                "match": {
+                    "job_name.keyword": f"{job}"
+                }
+            }
+            results = self.__query_get_hits(
+                query=query_body,
+                index='logstash_jenkins'
+            )
+
+            # We need to process all the results because
+            # in the same build we can have different tests
+            # So we can't use '"size": 1' in the query
+            if results:
+                for result in results:
+                    hits.append(result)
+
+        if not results:
+            return jobs_found
+
+        for hit in hits:
+            job_name = hit['_source']['job_name']
+            build_number = hit['_source']['build_num']
+            test_name = hit['_source']['test_name']
+            test_status = hit['_source']['test_status']
+            class_name = hit['_source'].get(
+                'test_class_name',
+                None
+            )
+            # Some build is not parsed good and contains
+            # More info than a time in the field
+            try:
+                test_duration = float(hit['_source']['test_time'])*1000
+            except ValueError:
+                LOG.warning("'test_time' field is not well parsed in "
```

Comment:
This line should probably go to debug too, I did a query for just one job and I got 200 of this warnings. That's fine at debug level, but not sure if we should show that in a normal run

Reply:
Done.